src/web_server.cpp

```diff
@@ -393,6 +396,48 @@ handleDivertMode(MongooseHttpServerRequest *request){
   DBUGF("Divert Mode: %d", divertmode);
 }
 
+// -------------------------------------------------------------------
+// Change current shaper mode 0:disable (default), 1:Enable
+// url: /shaper
+// -------------------------------------------------------------------
+void
+handleCurrentShaper(MongooseHttpServerRequest *request) {
+  MongooseHttpServerResponseStream *response;
+  if(false == requestPreProcess(request, response, CONTENT_TYPE_TEXT)) {
+    return;
+  }
+  shaper.setState(request->getParam("shaper").toInt() == 1? true: false);
+
+  response->setCode(200);
+  response->print("Current Shaper state changed");
+  request->send(response);
+
+  DBUGF("CurrentShaper: %d", shaper.getState());
+}
+
+// -------------------------------------------------------------------
+// Save current shaper settings
+// url: /saveshaper
+// -------------------------------------------------------------------
+void
+handleSaveCurrentShaper(MongooseHttpServerRequest *request) {
+  MongooseHttpServerResponseStream *response;
+  if(false == requestPreProcess(request, response, CONTENT_TYPE_TEXT)) {
+    return;
+  }
+
+  bool qshaperenabled = isPositive(request->getParam("enable"));
+  int qshapermaxpwr = request->getParam("maxpwr").toInt();
+  String qshaperlivepwr =  request->getParam("livepwr");
+
+  config_save_current_shaper(qshaperenabled, qshapermaxpwr, qshaperlivepwr);
+
+  response->setCode(200);
+  response->print("saved");
+  request->send(response);
+}
+
+
```

Comment:
Should be no need foe explicit endpoints, the generic config endpoint can be used

Reply:
I've made it as a Service in the web UI, so i've tried to match what has been done with others. They don't seem to use /config endpoints there
![sshot](https://user-images.githubusercontent.com/488789/181462270-9b415352-5ceb-4e61-b647-d98954ad7736.jpg)

Reply:
sorry, I just got what you've asked. :)
I'm trying something.